crates/nu-cli/src/repl.rs

```diff
@@ -205,22 +244,39 @@ pub fn evaluate_repl(
 
         start_time = std::time::Instant::now();
 
+        line_editor = if config.use_ansi_coloring {
+            line_editor.with_hinter(Box::new({
+                let stack = stack.get_unique_mut();
+                let style_computer = StyleComputer::from_config(engine_state, stack);
+                // As of Nov 2022, "hints" color_config closures only get `null` passed in.
+                let style = style_computer.compute("hints", &Value::nothing(Span::unknown()));
+                CwdAwareHinter::default().with_style(style)
+            }))
+        } else {
+            line_editor.disable_hints()
+        };
+
```

Comment:
```suggestion
        let stack = Arc::new(unique_stack);
```
`Arc` creation can be delayed until here.

Reply:
Since this is in a loop, the creation of the `Arc` can't happen here, as `unique_stack` gets moved in the previous iteration of the loop.

I did try to move up the point of creation of stack to right before the loop. It removes a couple calls to `get_unique_mut`.

Reply:
Ok was able to remove all of them thanks to this. Thanks!